src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.J.cs

```diff
@@ -196,16 +198,17 @@ private static bool TryParseDateTimeOffsetJ(ReadOnlySpan<byte> source, out DateT
 						break;
 					}
 
-					fraction = prevFractionTimesTen + (int)(source[sourceIndex++] - (uint)'0');
+					fraction = prevFractionTimesTen + (int)(curByte - (uint)'0');
+					sourceIndex++;
 				}
 			}
 
 			if (fraction != 0)
 			{
-				int fractionTimesTen;
-				while ((fractionTimesTen = fraction * 10) <= Utf8Constants.MaxDateTimeFraction)
+				// Note this is 1 order of magnitude less than Utf8Constants.MaxDateTimeFraction
+				while (fraction <= 999_999)
```

Comment:
nit: use named constant for this. Maybe `MaxDateTimeFractionDiv10` or something.